tds-itemselection-impl/src/test/java/msb/MsbAssessmentSelectionServiceTest.java

```diff
@@ -0,0 +1,268 @@
+        Assert.assertTrue(filterredItemCandidates.get(1).getAlgorithm() == "fixedform");
+        Assert.assertTrue(filterredItemCandidates.get(0).getSegmentKey() == "(SBAC_PT)SBAC-MSB-IRP-Perf-MATH-7-Summer-2015-2016");
+        Assert.assertTrue(filterredItemCandidates.get(1).getSegmentKey() == "(SBAC_PT)SBAC-MSB-IRP-Perf-MATH-11-Summer-2015-2016");
+    }
+
+    @Test
+    public void filterItemCandidatesWithExcludedAlgorithmTypeFilterShouldReturnNoResults() {
+        // Arrange
+        List<ItemCandidatesData> itemCandidatesData = retrieveItemCandidatesData();
+
+        // Act
+        List<ItemCandidatesData> filterredItemCandidates = msbAssessmentSelectionService.filterItemCandidatesByAlgorithm(itemCandidatesData, "nonsense");
+
+        // Assert
+        Assert.assertTrue(filterredItemCandidates.isEmpty());
+    }
+
+    @Test
+    public void filterItemCandidatesWithNoCandidatesShouldReturnNoResults() {
+        // Arrange
+        List<ItemCandidatesData> itemCandidatesData = new ArrayList<>();
+
+        // Act
+        List<ItemCandidatesData> filterredItemCandidates = msbAssessmentSelectionService.filterItemCandidatesByAlgorithm(itemCandidatesData, "nonsense");
+
+        // Assert
+        Assert.assertTrue(filterredItemCandidates.isEmpty());
+    }
+
+    //endregion
+
+    //region getTestSegmentsForItemCandidates Tests
+
+    @Test
+    public void getTestSegmentsForItemCandidatesFourCandidatesReturnsFourSegments() throws Exception {
+        // Arrange
+        List<ItemCandidatesData> itemCandidatesData = retrieveItemCandidatesData();
+        SQLConnection connection = mock(SQLConnection.class);
+        SegmentCollection2 segmentCollection = SegmentCollection2.getInstance();
+
+        // Act
+        List<TestSegment> testSegments = msbAssessmentSelectionService.getTestSegmentsForItemCandidates(itemCandidatesData, segmentCollection, connection);
+
+        // Assert
+        Assert.assertNotNull(testSegments);
+        Assert.assertTrue(testSegments.size() == itemCandidatesData.size());
+        for(ItemCandidatesData itemCandidate : itemCandidatesData) {
+            boolean segmentMatch = false;
+            for(TestSegment testSegment : testSegments) {
+                if(testSegment.getSegmentKey().compareToIgnoreCase(itemCandidate.getSegmentKey()) == 0) {
+                    segmentMatch = true;
+                    break;
+                }
+            }
+            Assert.assertTrue(segmentMatch);
+        }
+    }
+
+    // endregion
+
+    // region buildCombinedItemGroups tests
+
+    @Test
+    @Ignore
+    public void buildCombinedItemGroupsValidTestSegmentsProducesOneGroupPerSegmentContainingAllSegmentQuestions() {
+        // Arrange
+        List<TestSegment> testSegments = retrieveTestSegmentData();
+
+        // Act
+        List<ItemGroup> result = msbAssessmentSelectionService.buildCombinedItemGroups(testSegments);
+
+        // Assert
```

Comment:
No assertion here.  If we can't fix the test (or code it is testing) then we should remove the test.  Since it is new and @Ignore.